Zend/zend_string.h

```diff
@@ -339,6 +339,16 @@ static zend_always_inline void zend_string_release_ex(zend_string *s, bool persi
 	}
 }
 
+static zend_always_inline bool zend_string_equals_str(const zend_string *s1, const char *s2, size_t s2_length)
+{
+	return ZSTR_LEN(s1) == s2_length && !memcmp(ZSTR_VAL(s1), s2, s2_length);
+}
+
+static zend_always_inline bool zend_string_equals_cstr(const zend_string *s1, const char *s2)
+{
+	return zend_string_equals_str(s1, s2, strlen(s2));
+}
```

Comment:
I called it `_cstr` because it accepts a C string (i.e. null-terminated). If I pass a `size_t`, it's not a C string.
I don't quite understand what you want me to do, but I did find the existing nomenclature confusing. From the name of a function, I couldn't deduce what kind of parameters it accepts. In this PR, I tried my best to adapt to whatever naming was there.